### ISSUE TITLE
Counter does not reset on flush?

### ISSUE THREAD

**Issue by didip** [cyberdelia/go-metrics-graphite#5]
Unlike statsd, it looks like counter does not reset on every flush.

Is this expected?

**Comment by dmowcomber** [comment]
I would also like to know if this is expected behavior.

**Comment by m1stermanager** [comment]
+1 on this. I would expect the counter to reset and therefore have to write my own once per second sync function

**Comment by m1stermanager** [comment]
@didip @dmowcomber @cyberdelia would we expect this behavior to perhaps be part of the Config struct or should it always clear a counter on flush?

**Comment by didip** [comment]
Config struct is much better.

**Comment by dmowcomber** [comment]
It would be a breaking change if it suddenly was resetting the counter. I think it should be defined in a config with it defaulting to false.

**Comment by m1stermanager** [comment]
i agree completely. i'll try to get to this one because it really shouldn't be horrible. any ideas on the setting name? something like `ResetCountersOnFlush` or something similar?

**Comment by didip** [comment]
That's not bad, it's quite descriptive.

**Comment by buzzdan** [comment]
any updates on this issue ?
@m1stermanager ? @dmowcomber  ?

**Comment by m1stermanager** [comment]
@buzzdan i never actually had the time to do this but feel free to take a stab. If I do get the time I wouldn't mind trying it but i'd be happy to see you beat me to it
